.github/workflows/automatic-doc-checks.yml

```diff
@@ -0,0 +1,66 @@
+name: Automatic documentation checks
```

Comment:
Not that I'm against GH workflows, I actually prefer them, but there is already at least one Sphinx linter in the top-level Makefile, running in a different GH workflow. So I'd rather see everything somewhat unified. Example: if I do `make tests` right now, I won't get spelling nor linkcheck.